core/src/test/scala/integration/kafka/api/TransactionsTest.scala

```diff
@@ -218,6 +224,8 @@ class TransactionsTest extends IntegrationTestHarness {
     producer1.abortTransaction()
     producer2.commitTransaction()
 
+    maybeWaitForAtLeastOneSegmentUpload(new TopicPartition(topic1, 0))
```

Comment:
should we also assert that no segment in remote is present if the transaction is not closed (committed or aborted)? This will test the fact that we don't upload before least stable offset which only increments after commit/abort.

Reply:
Good point! Can we take this up while fixing the `testFencingOnTransactionExpiration` test?

Reply:
@showuon while you are fixing the `testFencingOnTransactionExpiration` please take up this comment as well.

Reply:
Will do